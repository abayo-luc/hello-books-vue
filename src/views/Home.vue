<template>
  <div class="home">
    <DashCards />
    <div class="content">
      <div class="all-books">
        <books-row v-for="(row, index) in rows" :key="index" :books="row" />
      </div>
      <div class="right-reading-list">
        <reading-list />
      </div>
    </div>
  </div>
</template>

<script>
import axios from 'axios';
import DashCards from '../components/DashCards.vue';
import BooksRow from '../components/BooksRow.vue';
import { D3transform } from '../utils/data';
import ReadingList from '../components/ReadingList.vue';

export default {
  name: 'home',
  components: {
    DashCards,
    BooksRow,
    ReadingList,
  },
  data() {
    return {
      rows: [],
      errors: [],
    };
  },
  created() {
    axios
      .get('https://jsonplaceholder.typicode.com/photos?_limit=18')
      .then((res) => {
        this.rows = [...D3transform(res.data)];
      })
      .catch(err => this.error.push(err.message));
  },
};
</script>
<style lang="scss" scoped>
.content {
  margin-top: 5em;
  display: flex;
  flex-direction: row;
  .all-books {
    width: 80%;
  }
  .right-reading-list {
    width: 20%;
  }
}
</style>
