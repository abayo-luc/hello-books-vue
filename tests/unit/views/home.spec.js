import {
  shallowMount,
} from '@vue/test-utils';
import Home from '../../../src/views/Home.vue';


describe('Home Page', () => {
  it('should match the snapshot', async () => {
    const wrapper = await shallowMount(Home, {
      stubs: ['router-link', 'router-view'],
      mocks: {
        $route: {
          meta: 'main-layout',
        },
      },
    });
    expect(wrapper).toMatchSnapshot();
  });
});
