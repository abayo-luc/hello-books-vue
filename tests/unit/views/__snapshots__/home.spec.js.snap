// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page should match the snapshot 1`] = `
<div class="home">
  <dashcards-stub></dashcards-stub>
  <div class="content">
    <div class="all-books"></div>
    <div class="right-reading-list">
      <reading-list-stub></reading-list-stub>
    </div>
  </div>
</div>
`;
